couchdb/tests/tests.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -e
+WAIT_THRESHOLD="${WAIT_THRESHOLD:-12}"
+SLEEP_SECONDS="${SLEEP_SECONDS:-5}"
+
+
+check_if_couchdb_cluster_is_ready(){
+
+
+  wait_count=0
+  until curl -s http://$COUCHDB_USER:$COUCHDB_PASSWORD@$NODENAME:5984/_cluster_setup | grep "cluster_finished" > /dev/null
+  do
+    echo "Waiting for cht couchdb cluster to be set up" >&2
+    wait_count=$((wait_count +1))
+    if [[ "$wait_count" -gt $WAIT_THRESHOLD ]]; then
+      echo "Couchdb clustering failed" >&2
+      exit 1
+    fi
+    sleep $SLEEP_SECONDS
+  done
+  echo "couchdb cluster is ready">&2
+
+}
+
+
+
+case $TEST_MODE in
+
+CLUSTERED)
+check_if_couchdb_cluster_is_ready
+/app/tests/bats/bin/bats /app/tests/tests.bats
```

Comment:
I took a look into that GH action, unfortunately it does not include `bats-support`, `bats-assert`, and `bats-shell-mock`, which we [load](https://github.com/medic/cht-core/blob/767e1edeac81bd5e20d3ee552fedb940d294b695/couchdb/tests/tests.bats#L2).

I did find another [GH Action](https://hub.docker.com/r/ffurrer/bats) that includes some of those add-ons, but not bats-shell-mock, which is [unsupported from capitolone](https://github.com/capitalone/bash_shell_mock#due-to-changes-in-the-priorities-this-project-is-currently-not-being-supported-the-project-is-archived-as-of-91721-and-will-be-available-in-a-read-only-state-please-note-since-archival-the-project-is-not-maintained-or-reviewed). I'm going to try rip out bats-shell-mock quickly and perhaps we can use this GH action after that is completed.

Reply:
I removed the git submodules and we are now using a bats docker image to run our tests! It's not part of GH actions, still manually. We should be able to port that over to GH Actions pretty easily in a latter PR.